# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/AudioSinkWrapper.cpp
# Commit: 16633fa96586
# Full Hash: 16633fa96586cb01af6dbfb03d062f1b32691ae9
# Author: Paul Adenot <paul@paul.cx>
# Date: 2022-05-24 21:44:48
# Regressor Bug: 1743834
# File Overlap Count: 1
# Description:
#   Bug 1743834 - Split initialization and start in AudioSinkWrapper. r=alwu
#   
#   No change in functionality, this is to prepare for the next patch that will do
#   the initialization asynchronously because it's expensing and blocking.
#   
# ==============================================================================

diff -r 505d7df63978 -r 16633fa96586 dom/media/mediasink/AudioSinkWrapper.cpp
--- a/dom/media/mediasink/AudioSinkWrapper.cpp	Tue May 24 13:09:09 2022 +0000
+++ b/dom/media/mediasink/AudioSinkWrapper.cpp	Tue May 24 13:09:09 2022 +0000
@@ -301,8 +301,13 @@
 
   if (!IsMuted()) {
     LOG("Not muted: starting a new audio sink");
-    mAudioSink.reset(mCreator->Create(aStartTime));
-    rv = mAudioSink->Start(mParams, mEndedPromiseHolder);
+    mAudioSink.reset(mCreator->Create());
+    rv = mAudioSink->InitializeAudioStream(mParams);
+    if (NS_FAILED(rv)) {
+      LOG("AudioSink initialization failure");
+      return rv;
+    }
+    rv = mAudioSink->Start(aStartTime, mEndedPromiseHolder);
   } else {
     LOG("Muted: not starting an audio sink");
   }